src/Symfony/Component/Messenger/Event/AbstractWorkerMessageEvent.php

```diff
@@ -15,7 +15,7 @@
 
 abstract class AbstractWorkerMessageEvent
 {
-    private $envelope;
+    protected $envelope;
```

Comment:
This is not needed - we can use `$this->getEnvelope()` already

Reply:
To get the envelope, yes. But we need to override `$envelope` to add the stamp in (as `Envelope` is immutable). `Envelope->with()` clones the envelope and returns the result.

Reply:
👍  reverted back to the `addStamps()` method on the abstract parent.